semantic_router/schema.py

```diff
@@ -25,7 +25,7 @@ class EncoderInfo(BaseModel):
 
 class RouteChoice(BaseModel):
     name: Optional[str] = None
-    function_call: Optional[dict] = None
+    function_call: Optional[list[dict]] = None
```

Comment:
The output choice should still only be one function call no?

Reply:
No. OpenAI's Function calling allows for multiple functions to be called based on a single user utterance. 

E.g. see here: https://cookbook.openai.com/examples/fine_tuning_for_function_calling

They have utterances 

```
straightforward_prompts = ['Land the drone at the home base',
 'Take off the drone to 50 meters',
 'change speed to 15 kilometers per hour',
  'turn into an elephant!']

```

which result in:

```
Land the drone at the home base
FunctionCall(arguments='{\n  "location": "home_base"\n}', name='land_drone') 

Take off the drone to 50 meters
FunctionCall(arguments='{\n  "altitude": 50\n}', name='takeoff_drone') 

change speed to 15 kilometers per hour
FunctionCall(arguments='{ "speed": 15 }', name='set_drone_speed') 

turn into an elephant!
FunctionCall(arguments='{}', name='reject_request') 
```

I figured if we're going to make multiple functions available to a single route, we may as well use OpenAIs full functionality and allow multiple to be called at once. I'm implementing the same in the original prompt based method (for other LLMs), it now outputs JSON with details for calling multiple functions at a time.

Reply:
but these are not utterances, they are four separate prompts? Our route calls only consume a single query, so they should only trigger a single function call?

Reply:
You're correct about the above example, I misread it. However, it is possible to obtain multiple function arguments from a single user input utterance (I've tested this with both the OpenAI Function Calling feature, and our custom prompt engineering method).